src/main/java/org/jabref/gui/preferences/AppearancePrefsTab.java

```diff
@@ -10,58 +12,151 @@
 import javafx.scene.layout.VBox;
 
 import org.jabref.gui.DialogService;
+import org.jabref.gui.JabRefFrame;
 import org.jabref.gui.util.ControlHelper;
 import org.jabref.logic.l10n.Localization;
 import org.jabref.preferences.JabRefPreferences;
 
+import java.io.*;
+import java.net.URISyntaxException;
+
 class AppearancePrefsTab extends Pane implements PrefsTab {
 
     private final JabRefPreferences prefs;
     private final CheckBox fontTweaksLAF;
+    private final CheckBox accessDarkThemes;
     private final TextField fontSize;
     private final CheckBox overrideFonts;
     private final VBox container = new VBox();
     private final DialogService dialogService;
+    private static final String DEFAULT_PATH_MAIN_CSS= JabRefFrame.class.getResource("Base.css").getPath();
+    private static final String DARK_THEME_PATH_CSS= JabRefFrame.class.getResource("Dark.css").getPath();
+    private static final String DEFAULT_PREF_MAIN_CSS= PreferencesDialog.class.getResource("PreferencesDialog.css").getPath();
+    private static final String DARK_PREF_MAIN_CSS= JabRefFrame.class.getResource("DarkPrefBox.css").getPath();
 
     /**
      * Customization of appearance parameters.
      *
      * @param prefs a <code>JabRefPreferences</code> value
      */
+
     public AppearancePrefsTab(DialogService dialogService, JabRefPreferences prefs) {
         this.dialogService = dialogService;
         this.prefs = prefs;
 
         overrideFonts = new CheckBox(Localization.lang("Override default font settings"));
+        accessDarkThemes=new CheckBox(Localization.lang("Access Dark Theme"));
         fontSize = new TextField();
         fontSize.setTextFormatter(ControlHelper.getIntegerTextFormatter());
         Label fontSizeLabel = new Label(Localization.lang("Font size:"));
         HBox fontSizeContainer = new HBox(fontSizeLabel, fontSize);
         VBox.setMargin(fontSizeContainer, new Insets(0, 0, 0, 35));
         fontSizeContainer.disableProperty().bind(overrideFonts.selectedProperty().not());
+        fontSizeContainer.disableProperty().bind(accessDarkThemes.selectedProperty().not());
         fontTweaksLAF = new CheckBox(Localization.lang("Tweak font rendering for entry editor on Linux"));
-
-        container.getChildren().addAll(overrideFonts, fontSizeContainer, fontTweaksLAF);
+        container.getChildren().addAll(overrideFonts,fontSizeContainer, fontTweaksLAF,accessDarkThemes);
 
     }
 
     public Node getBuilder() {
         return container;
     }
 
+    //Basically replace each of the files with each other
+    //Ultimately you have to read the contents of the file and replace it
+    private void addLogic(CheckBox accessDarkTheme) throws URISyntaxException {
+
+        if(accessDarkTheme.isSelected()==true){
+
+            File basecaseFile= new File(DEFAULT_PATH_MAIN_CSS);
+            File darkthemecssFile= new File(DARK_THEME_PATH_CSS);
```

Comment:
I don't understand why the copy and replace is necessary. I thought it's just pointing the JavaFX loader to the new path 🤔